polynote-frontend/polynote/ui.js

```diff
@@ -2112,6 +2128,30 @@ export class MainUI extends EventTarget {
         }
     }
 
+    handleHashChange(evt) {
+        // TODO: we need a better way to tell whether the notebook has been rendered rather than just using setTimeout and hoping the timing will work.
+        setTimeout(() => {
+            const hash = document.location.hash;
+            // the hash can (potentially) have two parts: the selected cell and selected lines.
+            // for example: #Cell2,6-12 would mean Cell2 lines 6-12
+            const [hashId, lines] = hash.slice(1).split(",");
+
+            const selected = document.getElementById(hashId);
+            if (selected && selected.cell && selected.cell !== Cell.currentFocus) {
+
+                // highlight lines
+                if (lines) {
+                    let [startLine, endLine] = lines.split("-");
+                    if (!endLine) endLine = startLine;
+                    selected.cell.editor.setSelection(new monaco.Selection(parseInt(startLine), 0, parseInt(endLine), 0))
```

Comment:
Let's use the highlight mechanism rather than selection. Like what execution does?

Reply:
Done! It looks a lot nicer as a highlight - though one downside is that the lines are permanently highlighted. 

What do you think about clearing the highlight when a cell loses focus (as long as it isn't running)?